audio_analysis.go

```diff
@@ -0,0 +1,119 @@
+	Confidence float64 `json:"confidence"`
+}
+
+type AudioAnalysisMeta struct {
+	AnalyzerVersion string  `json:"analyzer_version"`
+	Platform        string  `json:"platform"`
+	DetailedStatus  string  `json:"detailed_status"`
+	StatusCode      int     `json:"status"`
+	Timestamp       int64   `json:"timestamp"`
+	AnalysisTime    float64 `json:"analysis_time"`
+	InputProcess    string  `json:"input_process"`
+}
+
+type Section struct {
+	Measure
+	Loudness                float64 `json:"loudness"`
+	Tempo                   float64 `json:"tempo"`
+	TempoConfidence         float64 `json:"tempo_confidence"`
+	Key                     int     `json:"key"`
+	KeyConfidence           float64 `json:"key_confidence"`
+	Mode                    int     `json:"mode"`
+	ModeConfidence          float64 `json:"mode_confidence"`
+	TimeSignature           int     `json:"time_signature"`
+	TimeSignatureConfidence float64 `json:"time_signature_confidence"`
+}
+
+type Segment struct {
+	Measure
+	LoudnessStart   float64   `json:"loudness_start"`
+	LoudnessMaxTime float64   `json:"loudness_max_time"`
+	LoudnessMax     float64   `json:"loudness_max"`
+	LoudnessEnd     float64   `json:"loudness_end"`
+	Pitches         []float64 `json:"pitches"`
+	Timbre          []float64 `json:"timbre"`
+}
+
+type AudioAnalysisTrack struct {
+	NumSamples              int64   `json:"num_samples"`
+	Duration                float64 `json:"duration"`
+	SampleMD5               string  `json:"sample_md5"`
+	OffsetSeconds           int     `json:"offset_seconds"`
+	WindowSeconds           int     `json:"window_seconds"`
+	AnalysisSampleRate      int64   `json:"analysis_sample_rate"`
+	AnalysisChannels        int     `json:"analysis_channels"`
+	EndOfFadeIn             float64 `json:"end_of_fade_in"`
+	StartOfFadeOut          float64 `json:"start_of_fade_out"`
+	Loudness                float64 `json:"loudness"`
+	Tempo                   float64 `json:"tempo"`
+	TempoConfidence         float64 `json:"tempo_confidence"`
+	TimeSignature           int     `json:"time_signature"`
+	TimeSignatureConfidence float64 `json:"time_signature_confidence"`
+	Key                     Key     `json:"key"`
+	KeyConfidence           float64 `json:"key_confidence"`
+	Mode                    Mode    `json:"mode"`
+	ModeConfidence          float64 `json:"mode_confidence"`
+	CodeString              string  `json:"codestring"`
+	CodeVersion             float64 `json:"code_version"`
+	EchoprintString         string  `json:"echoprintstring"`
+	EchoprintVersion        float64 `json:"echoprint_version"`
+	SynchString             string  `json:"synchstring"`
+	SynchVersion            float64 `json:"synch_version"`
+	RhythmString            string  `json:"rhythmstring"`
+	RhythmVersion           float64 `json:"rhythm_version"`
+}
+
+// GetAudioAnalysis queries the Spotify web API for an audio analysis of a single track
+// If an object is not found, a nil value is returned in the appropriate position.
```

Comment:
This comment doesn't seem correct.  It only returns a single value, so the _appropriate position_ part is not relevant.